test/test_dynamics.jl

```diff
@@ -474,9 +413,55 @@ timesteppers = (:QuasiAdamsBashforth2, :RungeKutta3)
     end
 
     @testset "Internal wave" begin
-        for timestepper in (:QuasiAdamsBashforth2,) #timesteppers
-            @info "  Testing internal wave [$timestepper]..."
-            @test internal_wave_test(timestepper)
+        include("test_internal_wave_dynamics.jl")
+
+        Nx = Nz = 128
+        Lx = Lz = 2π
+
+        # Regular grid with no flat dimension
+        regular_grid = RegularRectilinearGrid(topology=(Periodic, Periodic, Bounded),
+                                              size=(Nx, 1, Nz), x=(0, Lx), y=(0, Lx), z=(-Lz, 0))
+
+        # Regular grid with a flat y-dimension
+        flat_grid = RegularRectilinearGrid(topology=(Periodic, Flat, Bounded),
+                                           size=(Nx, Nz), x=(0, Lx), z=(-Lz, 0))
+
+        # Vertically stretched grid with regular spacing and no flat dimension
+        z_faces = collect(znodes(Face, regular_grid))
+        unstretched_grid = VerticallyStretchedRectilinearGrid(topology=(Periodic, Periodic, Bounded),
```

Comment:
Referring to `VerticallyStretchedGrid` as `unstretched_grid` might create confusion. I get why you did it, but might be worth to stop a bit and think of a better name. I think just `stretched_grid` is fine, even though in practice it's non-stretched.

Reply:
The notion I am trying to express is: "A regular grid implemented with VerticallyStretchedRectilinearGrid type."

Elsewhere in the code we use the name `unstretched_grid` to denote such a grid.

I think calling it a "stretched grid" is confusing, because the grid is not stretched.

I certainly appreciate suggestions for alternative names and will implement them if I think they are an improvement. If we find a better name we should also make that change elsewhere so that our naming is consistent.

Reply:
Well, all I can say is that it would definitely confuse me if I found this in the middle of the code. But if you think I'm a minority, then feel free to ignore my comment

Reply:
I agree that it's cryptic or terse. We could say `regular_stretched_grid`. I thought that was more confusing because it seemed to be a contradiction --- but perhaps you think this is better? I'm open to other suggestions as well. I just don't want to use `stretched_grid` because I was confused about this specifically in another context. I think `unstretched_grid` is better because, while cryptic and confusing, it is not misleading.

I think the actual problem is that we can build a `VerticallyStretchedRectilinearGrid` that isn't actually stretched. This illustrates the name `VerticallyStretchedRectilinearGrid` is imperfect I think. I think in the long run we should have just one grid type, `RectilinearGrid`, which can be regular (constant grid metrics) or "possibly stretched" (array or function metrics for any direction). But that's a separate issue.

Reply:
I've used more verbose and specific names for all the grids. Let me know if this is an improvement.

Reply:
It's indeed much more verbose. But I personally think it's an improvement and I'm happy with it!

Reply:
It's indeed much more verbose. I personally think it's an improvement and I'm happy with it!